include/logging/log_core.h

```diff
@@ -283,12 +326,51 @@ static inline char z_log_minimal_level_to_char(int level)
 		}							       \
 	} while (false)
 
+
+#define __LOG_VA(_level, _id, _filter, _format, _valist, _argnum, _strdup) \
+		do {									   \
+			bool is_user_context = _is_user_context();			   \
+											   \
+			if (Z_LOG_CONST_LEVEL_CHECK(_level)) {				   \
+				if (IS_ENABLED(CONFIG_LOG_MINIMAL)) {			   \
+					if (IS_ENABLED(CONFIG_LOG_PRINTK)) { \
+						log_printk(_format, _valist); \
+					} else { \
+						vprintk(_format, _valist); \
+					} \
+				} else if (is_user_context ||				   \
+					   (_level <= LOG_RUNTIME_FILTER(_filter))) {  \
+					struct log_msg_ids src_level = {		   \
+						.level = _level,			   \
+						.domain_id = CONFIG_LOG_DOMAIN_ID,	   \
+						.source_id = _id			   \
+					};						   \
+						__LOG_INTERNAL_VA(is_user_context, 	   \
+								   src_level,		   \
+								   _format, _valist, _argnum, _strdup); \
+				}							   \
+			}								   \
+			if (false) {							   \
+				/* Arguments checker present but never evaluated.*/    \
+				/* Placed here to ensure that __VA_ARGS__ are*/ 	   \
+				/* evaluated once when log is enabled.*/		   \
+				/*log_printf_arg_checker(__VA_ARGS__);*/			   \
+			}								   \
+		} while (false)
+
+
 #define Z_LOG(_level, ...)			       \
 	__LOG(_level,				       \
 	      (u16_t)LOG_CURRENT_MODULE_ID(),	       \
 	      LOG_CURRENT_DYNAMIC_DATA_ADDR(),	       \
 	      __VA_ARGS__)
 
+#define Z_LOG_VA(_level, _format, _valist, _argnum, _strdup)			       \
```

Comment:
`_str` is used instead of `_format` in this file. I would keep it for consistency.

Reply:
i wonder if argnum is needed here? Couldn't `__LOG_INTERNAL_VA` call it since it is needed only there?

Reply:
Macro OPENTHREAD_CONFIG_PLAT_LOG_FUNCTION__COUNT_ARGS counts arguments passed to otPlatLog function, so we have it for free at compile time.